units/provisioning_setup_host.go

```diff
@@ -636,8 +586,8 @@ func (j *setupHostJob) fetchRemoteTaskData(ctx context.Context) error {
 	return errors.Wrap(err, "could not fetch remote task data")
 }
 
-func shouldRetryProvisioning(h *host.Host) bool {
-	return h.ProvisionAttempts < provisionRetryLimit && h.Status == evergreen.HostProvisioning && !h.Provisioned
+func (j *setupHostJob) shouldRetryProvisioning() bool {
+	return j.RetryInfo().GetRemainingAttempts() > 0 && j.host.Status == evergreen.HostProvisioning && !j.host.Provisioned
```

Comment:
Do we need to check remaining attempts? I thought the underlying amboy retry logic would take care of that for us.

Reply:
Checking the provisioning attempt is only a safeguard because whether the job retries depends heavily on the host status and I don't want to deal with fixing this messy code. There are 2 cases to consider when the job returns an error:

1. `provisionHost` keeps the host in the provisioning state and returns an error. In that case, the job can still retry, so `shouldRetryProvisioning` is true and a retryable error is added in the if case.
2. `provisionHost` sets the host to unprovisioned and returns an error. In that case, it just adds an unretryable error in the else case.

The assumption is the `provisionHost` will return an error in all return cases _except_ for the success case (i.e. the host is fully provisioned and set to running).

I renamed the method to make it a bit clearer. I also added a check that the host must be in the provisioning state to run this job.